v-next/hardhat/src/internal/builtin-plugins/node/artifacts/build-info-watcher.ts

```diff
@@ -0,0 +1,82 @@
+import type { FSWatcher } from "chokidar";
+
+import path from "node:path";
+
+import debug from "debug";
+
+export type BuildInfoWatcher = FSWatcher;
+export type BuildInfoHandler = (buildId: string) => Promise<void>;
+
+const log = debug("hardhat:core:tasks:node:artifacts");
+
+const STABILITY_THRESHOLD = 250;
+const POLL_INTERVAL = 50;
+
+/**
+ * `listener` is a callback function invoked when a new file is added to the
+ * build info directory. It processes the received event to check whether it is
+ * for a build info file.
+ *
+ * If so, it extracts the build id from it and triggers the provided handler
+ * with the extracted build id as an argument. Any errors encountered during
+ * handler execution are captured and logged.
+ *
+ * @param absolutePath - The absolute path of the file added to the build info directory.
+ *
+ * This function is exposed for testing purposes only.
+ */
+export async function listener(
+  handler: BuildInfoHandler,
+  absolutePath: string,
+): Promise<void> {
+  log(`Detected change in ${absolutePath}`);
+
+  const isBuildInfoFile =
+    absolutePath.endsWith(".json") && !absolutePath.endsWith(".output.json");
+
+  if (!isBuildInfoFile) {
+    log(`File ${absolutePath} is not a build info file`);
+    return;
+  }
+
+  const buildId = path.basename(absolutePath).replace(".json", "");
```

Comment:
Is there any implication from the change to include the solc version in the file name?
Do we consider the version to be part of the build id now?